src/lib/com/yetanalytics/xapipe/job/json.clj

```diff
@@ -0,0 +1,109 @@
+(ns com.yetanalytics.xapipe.job.json
+  "JSON Serialization/Deserialization for Jobs"
+  (:require [cheshire.core :as json]
+            [clojure.spec.alpha :as s]
+            [clojure.spec.gen.alpha :as sgen]
+            [cognitect.transit :as transit]
+            [com.yetanalytics.xapipe.job :as job]
+            [com.yetanalytics.xapipe.job.config :as config]
+            [xapi-schema.spec :as xs])
+  (:import [java.io ByteArrayInputStream ByteArrayOutputStream]))
+
+(s/fdef write-transit-str
+  :args (s/cat :data any?)
+  :ret string?)
+
+(defn write-transit-str
+  [data]
+  (with-open [baos (ByteArrayOutputStream.)]
+    (let [w (transit/writer baos :json)]
+      (transit/write w data)
+      (.toString baos "UTF-8"))))
+
+(s/fdef read-transit-str
+  :args (s/cat :t-str
+               (s/with-gen string?
+                 (fn []
+                   (sgen/fmap
+                    write-transit-str
+                    (s/gen any?)))))
+  :ret any?)
+
+(defn read-transit-str
+  [^String t-str]
+  (with-open [bais (ByteArrayInputStream. (.getBytes t-str "UTF-8"))]
+    (let [r (transit/reader bais :json)]
+      (transit/read r))))
+
+(defn- pack-paths
```

Comment:
I like how this was generalized for `-in` fns. could be useful again someday for partially-readable serialization